### PR TITLE
internal/identity: remove identity package

### PR DESCRIPTION
**Pull request by mhilton**
Change the custom identity client package for the standard one at
github.com/juju/idmclient.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1648/
Test FAILed.

File: internal/v5/auth_test.go [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ func (s *authSuite) TestGroupsForUserWithInvalidIdentityURL(c *gc.C) {
 	h := s.handler(c)
 	defer h.Close()
 	groups, err := h.GroupsForUser("someone")
-	c.Assert(err, gc.ErrorMatches, `cannot get groups for someone: cannot GET \"/v1/u/someone/groups\": cannot create request for \":::::/v1/u/someone/groups\": parse :::::/v1/u/someone/groups: missing protocol scheme`)
+	c.Assert(err, gc.ErrorMatches, `cannot parse ":::::": parse :::::: missing protocol scheme`)
```
Comment (fabricematrat) [inline internal/v5/auth_test.go]: why is it so cryptic the error ?

File: internal/v5/auth_test.go [hunk 2]
```diff
@@ -1089,7 +1089,7 @@ func (s *authSuite) TestGroupsForUserWithInvalidBody(c *gc.C) {
 	s.idM.body = "bad"
 	s.idM.contentType = "application/json"
 	groups, err := h.GroupsForUser("someone")
-	c.Assert(err, gc.ErrorMatches, `cannot get groups for someone: cannot unmarshal response: invalid character 'b' looking for beginning of value`)
+	c.Assert(err, gc.ErrorMatches, `GET .*: invalid character 'b' looking for beginning of value`)
```
Comment (fabricematrat) [inline internal/v5/auth_test.go]: same question about the GET .\* a bit cryptic ?

**Comment by fabricematrat** [comment]
:+1: code :ok: no QA

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1649/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1650/
Test PASSed.

File: internal/v5/auth_test.go (hunk 2 above)
Comment (mhilton) [inline internal/v5/auth_test.go]: .\* will actually be https://api.jujucharms.com/identity/u/fabrice/groups or similar.

File: internal/v5/auth_test.go (hunk 1 above)
Comment (mhilton) [inline internal/v5/auth_test.go]: in this case it's quite a silly thing to do.

Comment (mhilton) [inline internal/v5/auth_test.go]: I've re-added the cannot get groups for bit to make them less cryptic.

**Comment by rogpeppe** [comment]
LGTM, loving all that red!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1651/
Test PASSed.

**Comment by urosj** [comment]
QA required, other then that ... LGTM.

**Comment by mhilton** [comment]
## QA

```
% putadminagent  -k 'X3Yj/aThvG20FoBhRAIX+JbFk300r9Roc2D78r/37iw=' charmstore grouplist@idm
user=charmstore@admin@idm
% curl http://admin:password@localhost:8081/v1/u/charmstore@admin@idm
{"username":"charmstore@admin@idm","external_id":"","fullname":"","email":"","gravatar_id":"","idpgroups":["grouplist@idm"],"owner":"admin@idm","public_keys":["X3Yj/aThvG20FoBhRAIX+JbFk300r9Roc2D78r/37iw="]}
% bhttp localhost:8080/v5/whoami
{
    User: "martin-hilton"
    Groups: [
        "cloud-green-qa"
        "yellow"
        "juju-gui"
        "juju-jitsu"
        "juju"
        "juju-gui-charmers"
        "charming-devs"
        "juju-reports-viewers"
    ]
}
```

**Comment by urosj** [comment]
@mhilton thanks!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1652/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/charmstore-merge/477

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**